ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -216,6 +246,34 @@ private MessageWriteSetStateInStore checkWriteSetStateInStore(MessageWriteSet me
     }
   }
 
+  /**
+   * Checks the used capacity of the store against the configured percentage thresholds to see if the store
+   * should be read-only or read-write
+   * @param totalCapacity total capacity of the store in bytes
+   * @param usedCapacity total used capacity of the store in bytes
+   */
+  private void checkCapacityAndUpdateHelix(long totalCapacity, long usedCapacity) {
+    if (clusterManagerWriteStatusDelegate != null) {
+
+      int threshold = config.storeReadOnlyEnableSizeThresholdPercentage;
+      int delta = config.storeReadWriteEnableSizeThresholdPercentageDelta;
+      double percentFilled = (index.getLogUsedCapacity() / (double) log.getCapacityInBytes()) * 100;
+
+      if (percentFilled > threshold && !replicaId.isSealed()) {
+        if (!clusterManagerWriteStatusDelegate.seal(replicaId)) {
+          metrics.sealSetError.inc();
+        }
+      } else if (percentFilled <= threshold - delta && replicaId.isSealed()) {
+        if (!clusterManagerWriteStatusDelegate.unseal(replicaId)) {
+          metrics.sealSetError.inc();
```

Comment:
since these are expected to be infrequent operations, perhaps log an error as well? (as you are using the same metric for both).

Reply:
I think @vgkholla and I were considering a possible scenario where it went crazy with sending seal / unseal updates, and were afraid of logs overwhelming the system, so that's why we went with a metric instead of log... maybe I can do two separate metrics (sealSetError and unsealSetError)?

Reply:
yeah if either unavailability or configuration/temp state (don't update helix right now) cause this to return `false`, we are looking at an error log growth rate of put + delete QPS. Besides, I don't think the error log can offer more info than what the metric does?

Reply:
If the metrics are separate, then yes the error logging is unnecessary.

Reply:
added separate metrics